test.sh

```diff
@@ -54,103 +30,140 @@ elif [[ $($ENGINE ps -q -f name="$CONTAINER_NAME" | wc -l) -eq 0 ]]; then
-  $RUN $PIP install -r requirements-py2.txt
-fi
+  if [[ $OS == centos && $OS_VERSION == 7 ]]; then
+    # Get a less ancient version of pip to avoid installing py3-only packages
+    $RUN $PIP install "pip>=9.0.0,<10.0.0"
+    # ...but ancient enough to allow uninstalling packages installed by distutils
 
-# install with RPM_PY_SYS=true to avoid error caused by installing on system python
-$RUN sh -c "RPM_PY_SYS=true $PIP install rpm-py-installer"
+    # Older versions of setuptools don't understand the environment
+    # markers used by docker-squash's requirements
+    $RUN $PIP install -U setuptools
+  fi
 
-$RUN $PIP install docker-squash
-$RUN $PYTHON setup.py install
+  # Install other dependencies for tests
 
-# Install packages for tests
-$RUN $PIP install -r tests/requirements.txt
+  # Install osbs-client dependencies based on specfile
+  # from specified git source (default: upstream master)
+  $RUN rm -rf /tmp/osbs-client
+  $RUN git clone --depth 1 --single-branch \
+      "${OSBS_CLIENT_REPO}" --branch "${OSBS_CLIENT_BRANCH}" /tmp/osbs-client
+  $RUN $BUILDDEP --define "with_python3 ${WITH_PY3}" -y /tmp/osbs-client/osbs-client.spec
+  # Run pip install with '--no-deps' to avoid compilation
+  # This would also ensure all the deps are specified in the spec
+  $RUN $PIP install --upgrade --no-deps --force-reinstall \
+      "git+${OSBS_CLIENT_REPO}@${OSBS_CLIENT_BRANCH}"
 
-# CentOS needs to have setuptools updates to make pytest-cov work
-if [[ $OS != "fedora" ]]; then $RUN $PIP install -U setuptools; fi
+  $RUN $PIP install --upgrade --no-deps --force-reinstall git+https://github.com/DBuildService/dockerfile-parse
+  if [[ $PYTHON_VERSION == 2* ]]; then
+    $RUN $PIP install git+https://github.com/release-engineering/dockpulp
+    $RUN $PIP install -r requirements-py2.txt
+  fi
+
+  # install with RPM_PY_SYS=true to avoid error caused by installing on system python
+  $RUN sh -c "RPM_PY_SYS=true $PIP install rpm-py-installer"
+
+  $RUN $PIP install docker-squash
+  $RUN $PYTHON setup.py install
+
+  # Install packages for tests
+  $RUN $PIP install -r tests/requirements.txt
+
+  # CentOS needs to have setuptools updates to make pytest-cov work
+  if [[ $OS != "fedora" ]]; then $RUN $PIP install -U setuptools; fi
+}
 
 case ${ACTION} in
 "test")
+  prep_linux
   TEST_CMD="pytest tests --cov atomic_reactor --cov-report html"
   ;;
 "pylint")
+  prep_linux
   # This can run only at fedora because pylint is not packaged in centos
   # use distro pylint to not get too new pylint version
   $RUN $PKG install -y "${PYTHON}-pylint"
   PACKAGES='atomic_reactor tests'
   TEST_CMD="${PYTHON} -m pylint ${PACKAGES}"
   ;;
 "bandit")
+  prep_linux
   $RUN $PIP install bandit
   TEST_CMD="bandit-baseline -r atomic_reactor -ll -ii"
   ;;
+"markdownlint")
+  $RUN gem install mdl
+  TEST_CMD="mdl -g ."
+  ;;
 *)
   echo "Unknown action: ${ACTION}"
   exit 2
   ;;
 esac
 
 # Run tests
-$RUN ${TEST_CMD} "$@"
+$RUN "${TEST_CMD}" "$@"
```

Comment:
Why this change?

P.S. it's breaking test execution

Reply:
Lame attempt to fix quoting issue